libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -145,8 +135,8 @@ internal class RouteRefreshController(
      * - uuid is not empty;
      * - uuid is not equal to [MapboxNativeNavigatorImpl.OFFLINE_UUID].
      */
-    private fun DirectionsRoute?.isUuidValidForRefresh(): Boolean =
-        this?.requestUuid()
+    private fun NavigationRoute?.isUuidValidForRefresh(): Boolean =
+        this?.directionsResponse?.uuid()
```

Comment:
This https://github.com/mapbox/mapbox-navigation-android/pull/5411/files#r793665254 or `NavigationRoute.getRoute().requestUuid()`. If not, we'd be disabling refresh to anyone using the legacy approach.